plugins/helpers/OpenfdaHelper.py

```diff
@@ -0,0 +1,67 @@
+        logger.info(download_entry)
+        download_url = download_entry['file']
+        download_description = download_entry['display_name']
+        download_src_filename = os.path.basename(urlparse(download_url).path)
+        download_dest_filename = "{}.zip".format(uuid.uuid4())
+        download_dest_path = os.path.join(self.output, download_dest_filename)
+        download_resource_key = "openfda-event-{}".format(download_dest_filename)
+        logger.info("Download OpenFDA FAERs '{}', from '{}', destination file name '{}'".format(download_description,
+                                                                                                download_url,
+                                                                                                download_src_filename))
+        download_resource = type('download_resource', (object,), {
+            'uri': download_url,
+            'unzip_file': True,
+            'resource': download_resource_key,
+            'output_filename': download_dest_filename,
+            'accept': 'application/zip'
+        })
+        # Download the file
+        downloader = DownloadResource(self.output)
+        downloader.execute_download(download_resource, 7)
+        # Expand the ZIP file
+        extracted_filelist = dict()
+        with zipfile.ZipFile(download_dest_path, 'r') as zipf:
+            for event_file in zipf.filelist:
+                unzip_filename = event_file.filename
+                unzip_dest_filename = "{}.jsonl".format(uuid.uuid4())
+                unzip_dest_path = os.path.join(self.output, unzip_dest_filename)
+                logger.info("Inflating event file '{}', CRC '{}'".format(unzip_filename, event_file.CRC))
+                with zipf.open(unzip_filename, 'r') as compressedf:
+                    with open(unzip_dest_path, 'w') as inflatedf:
+                        logger.info("Extracting event file results")
+                        event_data = json.load(compressedf)
+                        if event_data['results']:
+                            for result in event_data['results']:
+                                inflatedf.write(json.dumps(result))
+                        else:
+                            logger.warning(
+                                "NO EVENT DATA RESULTS for event file '{}', source URL '{}', description '{}'".format(
+                                    unzip_filename, download_url, download_description))
+                unzip_resource_key = "{}-{}".format(download_resource_key, unzip_dest_filename)
+                extracted_filelist[unzip_dest_filename] = {
+                    'resource': unzip_resource_key,
+                    'gs_output_dir': unzip_dest_path
+                }
+        logger.warning("Removing processed ZIP file '{}'".format(download_dest_path))
+        os.remove(download_dest_path)
+        return extracted_filelist
+
+
+if __name__ == '__main__':
```

Comment:
This code doesn't seem to do anything.

Reply:
I removed a part of code that doesn't do anything useful.
I leave main because multiprocess requires a main in order to work.